openmetrics-api/src/main/java/io/airlift/openmetrics/types/Summary.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.openmetrics.types;
+
+import java.util.Map;
+
+import static io.airlift.openmetrics.ExpositionHelper.typeLine;
+import static io.airlift.openmetrics.ExpositionHelper.valueLine;
+
+public record Summary(Long count, Double sum, Double created, Map<Double, Double> quantiles)
+        implements Metric
+{
+    @Override
+    public CharSequence getMetricExposition(String metricName)
```

Comment:
cruious: why `CharSequence` and not `String`? Performance? I am not a big fan of returning mutable object, even if hidden behind interface. Maybe this is fine.